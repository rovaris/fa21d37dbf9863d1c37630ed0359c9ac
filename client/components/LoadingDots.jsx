// @flow
import React from 'react';
import { uniqueId } from 'lodash';

type LoadingDotsType = {
    numberOfDots: number,
};

const renderDots = (numberOfDots: number) => (
    [...Array(numberOfDots)].map(i => (
        <div className="circle" key={ uniqueId(i) } />
    ))
);

const LoadingDots = (props: LoadingDotsType) => (
    <div className="loader">
        { renderDots(props.numberOfDots) }
    </div>
);

export default LoadingDots;
