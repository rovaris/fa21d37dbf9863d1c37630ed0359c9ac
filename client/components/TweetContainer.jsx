// @flow
import React from 'react';
import { Container, Panel } from 'muicss/react';

type TweetContainerType = {
    text: string,
};

const TweetContainer = ({ text }):TweetContainerType => (
    <Container className="tweet-container">
        <Panel>
            <p>{ text }</p>
        </Panel>
    </Container>
);


export default TweetContainer;
