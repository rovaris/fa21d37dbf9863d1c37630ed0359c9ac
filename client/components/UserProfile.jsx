// @flow
import React from 'react';
import cx from 'classnames';

type UserProfileType = {
    name: string,
    profile: string,
    imgUrl: string,
    className?: string,
};

const UserProfile = ({ name, profile, imgUrl, className }):UserProfileType => (
    <div className={ cx('user-profile', className) }>
        <div>
            <img alt="profile" src={ imgUrl } />
        </div>
        <div className="text">
            <h2> { name } </h2>
            <p> {`@${profile}`} </p>
        </div>
    </div>
);

export default UserProfile;
