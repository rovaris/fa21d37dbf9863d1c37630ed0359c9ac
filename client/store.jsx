import { applyMiddleware, compose, createStore, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import createSagaMiddleware from 'redux-saga';

const sagaMiddleware = createSagaMiddleware();

const createAsyncStore = compose(
    applyMiddleware(thunk),
    applyMiddleware(sagaMiddleware),
    window.devToolsExtension ? window.devToolsExtension() : func => func,
)(createStore);

// sagaMiddleware.run(mySaga)

// export default createAsyncStore(
//     combineReducers(reducers),
// );

