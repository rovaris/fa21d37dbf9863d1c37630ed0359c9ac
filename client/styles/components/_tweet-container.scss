.tweet-container {
    & > div > p {
        word-wrap: break-word;
    }   
}