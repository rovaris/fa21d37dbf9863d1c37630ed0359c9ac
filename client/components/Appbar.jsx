// @flow
import React from 'react';
import { Appbar, Container, Button } from 'muicss/react';
import UserProfile from './UserProfile';

type AppBarType = {
    disconnect: () => void,
    name: string,
    profile: string,
    imgUrl: string,
};

const AppBar = ({ disconnect, name, profile, imgUrl }):AppBarType => (
    <Appbar>
        <Container>
            <UserProfile
                className="mui--pull-left mui--appbar-height"
                name={ name }
                profile={ profile }
                imgUrl={ imgUrl }
            />
            <div className="mui--pull-right mui--appbar-line-height">
                <Button
                    variant="raised"
                    color="secondary"
                    onClick={ disconnect }
                >
                    <span
                        style={ { paddingRight: '5px' } }
                        title="remove"
                        className="icon fa fa-sign-out"
                    />
                    Logout
                </Button>
            </div>
        </Container>
    </Appbar>
);

export default AppBar;
