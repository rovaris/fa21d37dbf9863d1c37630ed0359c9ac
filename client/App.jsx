import React from 'react';
import { connect } from 'react-redux';
import { Container, Panel } from 'muicss/react';
import { LoadingDots } from 'components';
import { LoginView, DashboardView } from './views';

const App = ({ reducer }) => {
    const { isLoggedIn, loading } = reducer;

    if (loading) {
        return (
            <Panel className="mui--text-center">
                <LoadingDots numberOfDots={ 3 } />
            </Panel>
        );
    }

    return (
        <Container>
            { isLoggedIn ?
                <DashboardView { ...reducer } /> :
                <LoginView />
            }
        </Container>
    );
};

const mapStateToProps = ({ LoginReducer }) => ({
    reducer: LoginReducer,
});

export default connect(mapStateToProps, null)(App);
