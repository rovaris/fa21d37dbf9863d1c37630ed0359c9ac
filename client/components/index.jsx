import LoadingDots from './LoadingDots';
import Appbar from './Appbar';
import TweetContainer from './TweetContainer';

export {
    LoadingDots,
    Appbar,
    TweetContainer,
};
