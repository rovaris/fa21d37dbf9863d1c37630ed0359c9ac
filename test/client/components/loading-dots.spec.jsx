import React from 'react';
import { LoadingDots } from '../../../client/components';
import renderer from 'react-test-renderer';

describe('Test Loading Dots behaviour', () => {

    function asserts(numberOfDots) {
        const component = renderer.create(
            <LoadingDots numberOfDots={ numberOfDots } />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        expect(tree.children.length).toBe(numberOfDots);

        tree.children.forEach(child => {
            expect(child.type).toBe('div');
            expect(child.props.className).toBe('circle');
            expect(child.children).toBeNull();
        });
    }

    it(`Scenario:
            Should render dots accordandly with the number of dots provided`, () => {
        asserts(3);
        asserts(6);
        asserts(8);
        asserts(10);
    });
});
