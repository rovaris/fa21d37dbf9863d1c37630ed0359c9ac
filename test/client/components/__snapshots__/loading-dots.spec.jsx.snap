// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loading Dots behaviour Scenario:
            Should render dots accordandly with the number of dots provided 1`] = `
<div
  className="loader"
>
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
</div>
`;

exports[`Test Loading Dots behaviour Scenario:
            Should render dots accordandly with the number of dots provided 2`] = `
<div
  className="loader"
>
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
</div>
`;

exports[`Test Loading Dots behaviour Scenario:
            Should render dots accordandly with the number of dots provided 3`] = `
<div
  className="loader"
>
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
</div>
`;

exports[`Test Loading Dots behaviour Scenario:
            Should render dots accordandly with the number of dots provided 4`] = `
<div
  className="loader"
>
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
  <div
    className="circle"
  />
</div>
`;
